tests/test_server.py

```diff
@@ -39,17 +39,35 @@ def heartbeat():
 
 def test_get_events(flask_client, bucket, benchmark):
     n_events = 100
+    start_time = datetime.now() - timedelta(1)
     for i in range(n_events):
         now = datetime.now() - timedelta(1)
         r = flask_client.post('/api/0/buckets/test/heartbeat?pulsetime=0'.format(bucket), json={'timestamp': now, 'duration': 0, 'data': {'random': random.random()}})
         assert r.status_code == 200
 
     @benchmark
     def get_events():
+        r = flask_client.get('/api/0/buckets/test/events'.format(bucket))
+        assert r.status_code == 200
+        assert r.json
+        assert len(r.json) == n_events
+
         r = flask_client.get('/api/0/buckets/test/events?limit=-1'.format(bucket))
         assert r.status_code == 200
         assert r.json
         assert len(r.json) == n_events
+        
+        r = flask_client.get('/api/0/buckets/test/events?limit=100'.format(bucket))
```

Comment:
The things which are good to test are:
- limit unspecified (which means all events, you test this correctly)
- limit specified as -1 (which means all events, you test this correctly)
- limit specified as something less than what's actually available (which means less events than what's available, **you do not currently test this**)
- limit specified as something more than what's available (which means all events, you test this correctly)

I think we should test to set limit to 10 here instead, then we would test all scenarios.

Reply:
Great! Thank you for the breakdown, I really appreciate it.